### PR TITLE
[UWP] Added mouse move support and fixes some mouse enter/leave scenarios

### PR DESCRIPTION
**Pull request by reseul**
1. Added "mouse move" support.
2. Change "mouse enter/leave/over/out" to rely mainly on "mouse move" events rather than on UIElement.PointerEnered/Exited. The latter are fired when OS decides the pointer enter/left what it thinks is the hit target, but that is wrong in some scenarios like "box-none" views overlapping non-related views.
The new mechanism basically makes all events consistent with the target resolution offered by the TouchHandler.GetReactViewTarget.

### CODE REVIEW

File: ReactWindows/ReactNative.Shared/UIManager/Events/TouchEventType.cs [hunk 1]
```diff
@@ -40,5 +40,10 @@ public enum TouchEventType
         /// Pointer exited event type.
         /// </summary>
         Exited,
+
+        /// <summary>
+        /// Pointer move event type.
+        /// </summary>
+        PointerMove
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/UIManager/Events/TouchEventType.cs]: `PointerMove` doesn't necessarily differentiate from `Move`, but `MoveWithoutStart` seems too verbose. `PointerMove` is probably fine (I can't think of anything better.

File: ReactWindows/ReactNative.Shared/UIManager/Events/TouchEventTypeExtensions.cs [hunk 2]
```diff
@@ -25,6 +25,8 @@ public static string GetJavaScriptEventName(this TouchEventType eventType)
                     return "topMouseOver";
                 case TouchEventType.Exited:
                     return "topMouseOut";
+                case TouchEventType.PointerMove:
+                    return "topMouseMoveCustom"; // Using a non-clashing name until this one propagates: https://github.com/facebook/react/commit/e96dc140599363029bd05565d58bcd4a432db370
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/UIManager/Events/TouchEventTypeExtensions.cs]: Basically we can change this once we update to `react-native@0.57` (and we have `react@16.4+`)

File: ReactWindows/ReactNative/Touch/TouchHandler.cs [hunk 3]
```diff
@@ -230,9 +357,44 @@ private void UpdatePointerForEvent(ReactPointer pointer, PointerPoint rootPoint,
...
-            pointer.AltKey = Window.Current.CoreWindow.GetKeyState(Windows.System.VirtualKey.Menu).HasFlag(CoreVirtualKeyStates.Down);
-            pointer.CtrlKey = Window.Current.CoreWindow.GetKeyState(Windows.System.VirtualKey.Control).HasFlag(CoreVirtualKeyStates.Down);
+            pointer.ShiftKey = keyModifiers.HasFlag(VirtualKeyModifiers.Shift);
+            pointer.AltKey = keyModifiers.HasFlag(VirtualKeyModifiers.Menu);
+            pointer.CtrlKey = keyModifiers.HasFlag(VirtualKeyModifiers.Control);
+        }
+
+        private static bool ShouldSendPointerMoveEvent(DependencyObject view)
+        {
+            // This is a bubbling event, so we have to check if mouse move hanlder is hooked to any ancestor
```
Comment (rozele) [inline ReactWindows/ReactNative/Touch/TouchHandler.cs]: `hanlder` -> `handler`

File: ReactWindows/ReactNative/Touch/TouchHandler.cs [hunk 4]
```diff
@@ -320,7 +482,69 @@ public override void Dispatch(RCTEventEmitter eventEmitter)
...
+                    }
+                }
+
+                if (_sendBubbled)
+                {
+                    eventEmitter.receiveEvent(ViewTag, EventName, eventData);
+                }
+            }
+        }
+
```
Comment (rozele) [inline ReactWindows/ReactNative/Touch/TouchHandler.cs]: Extra newline

File: ReactWindows/ReactNative/UIManager/BaseViewManager.cs [hunk 5]
```diff
@@ -311,40 +297,6 @@ private void OnSizeChanged(object sender, SizeChangedEventArgs e)
...
-                .Skip(1) // Skip the current view
-                .Any(v => v.GetPointerEvents() == PointerEvents.BoxOnly);
-
-            if (!hasBoxOnlyParent)
-            {
-                view.GetReactContext()
-                    .GetNativeModule<UIManagerModule>()
-                    .EventDispatcher
-                    .DispatchEvent(
-                        new PointerEnterExitEvent(TouchEventType.Entered, view.GetTag()));
```
Comment (rozele) [inline ReactWindows/ReactNative/UIManager/BaseViewManager.cs]: Delete `PointerEnterExitEvent` at the bottom of the file (unused?)

**Comment by reseul** [comment]
Closed by mistake, it lives again in https://github.com/Microsoft/react-native-windows/pull/1973

**Pull request closed without merging**